internal/servers/controller/handlers/targets/target_service.go

```diff
@@ -909,6 +1024,75 @@ func (s Service) removeInRepo(ctx context.Context, targetId string, hostSetIds [
 	return out, hs, cl, nil
 }
 
+func (s Service) addLibrariesInRepo(ctx context.Context, targetId string, libraryIds []string, libraries []*pbs.CredentialLibrary, version uint32) (target.Target, []*target.TargetSet, []*target.TargetLibrary, error) {
+	repo, err := s.repoFn()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	lids := make([]string, 0, len(libraryIds)+len(libraries))
+	lids = append(lids, libraryIds...)
+	for _, l := range libraries {
+		lids = append(lids, l.GetId())
+	}
+	out, hs, cl, err := repo.AddTargetCredentialLibraries(ctx, targetId, version, strutil.RemoveDuplicates(lids, false))
+	if err != nil {
+		// TODO: Figure out a way to surface more helpful error info beyond the Internal error.
+		return nil, nil, nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to add credential libraries to target: %v.", err)
+	}
+	if out == nil {
+		return nil, nil, nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to lookup target after adding credential libraries to it.")
+	}
+	return out, hs, cl, nil
+}
+
+func (s Service) setLibrariesInRepo(ctx context.Context, targetId string, libraryIds []string, libraries []*pbs.CredentialLibrary, version uint32) (target.Target, []*target.TargetSet, []*target.TargetLibrary, error) {
+	const op = "targets.(Service).setLibrariesInRepo"
+	repo, err := s.repoFn()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	lids := make([]string, 0, len(libraryIds)+len(libraries))
+	lids = append(lids, libraryIds...)
+	for _, l := range libraries {
+		lids = append(lids, l.GetId())
+	}
+	_, _, _, err = repo.SetTargetCredentialLibraries(ctx, targetId, version, strutil.RemoveDuplicates(lids, false))
+	if err != nil {
+		// TODO: Figure out a way to surface more helpful error info beyond the Internal error.
+		return nil, nil, nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to set credential libraries in target: %v.", err)
+	}
+
+	out, hs, cl, err := repo.LookupTarget(ctx, targetId)
```

Comment:
Why do a lookup here?  Doesn't the `SetTargetCredentialLibraries` return the same info?